src/protocols/secure_channel/DefaultSessionResumptionStorage.cpp

```diff
@@ -162,35 +162,37 @@ CHIP_ERROR DefaultSessionResumptionStorage::DeleteAll(FabricIndex fabricIndex)
         {
             continue;
         }
-        err       = LoadState(index.mNodes[cur], resumptionId, sharedSecret, peerCATs);
-        stickyErr = stickyErr == CHIP_NO_ERROR ? err : stickyErr;
+        err = LoadState(index.mNodes[cur], resumptionId, sharedSecret, peerCATs);
         if (err != CHIP_NO_ERROR)
         {
-            ChipLogError(SecureChannel,
-                         "Session resumption cache deletion partially failed for fabric index %u, "
-                         "unable to load node state: %" CHIP_ERROR_FORMAT,
-                         fabricIndex, err.Format());
```

Comment:
I'm confused.  These entries are in the index.  Isn't it unexpected for the entries not to be loadable?

Reply:
In an ideal world yes. However I am seeing it on several systems.

Reply:
That then indicates that the kvstore was in an inconsistent state.  Why isn't that worthy of logging?

Reply:
Here's a question: if we can't load an entry, we can't delete them.  How will they ever be deleted?  Isn't this a storage leak?

Reply:
Please see this comment: https://github.com/project-chip/connectedhomeip/pull/16741#discussion_r844632763

I think we need a very clear explanation of the state that existed in the kvstore to cause this logging.  Will the kvstore ever heal from this, or are we seeing condition that will persist forever?  If so, just burying the error logging is not reasonable.

As one data point, we chose NOT to use the SimpleSessionResumptionStorage implementation because it cannot be atomically updated and is, therefore, inherently prone to getting out of sync with itself.  I think what's reported here may be a result of that.

Reply:
Yes, it's a leak, but only if the entry actually exists (e.g. corrupt), in which case the error will not be "value not found" and the old code path will be executed.

Reply:
Can you be more specific?  There are three tables in the kvstore.  What was where to make this happen?

Reply:
The inconsistent state is created by chip-tool itself ours/days after deleting all /tmp/chip*

Reply:
No, I'm asking what the state of the three tables in the kvstore was, not what led to this.  I'm asking about the actual contents.

Given that the SimpleSessionResumptionStorage implementation can't do atomic updates, we would at least like to have some paths for potential self-healing.

Will the entries that the current code is complaining about in the log ever go away, or are these a permanent storage leak?  Can we ever reclaim slots in the index when this occurs?

Reply:
chip-tool does not seem to use kvstore for resumption, it uses chip_tool_config.ini. That's the one causing the problems.

Reply:
What is the code that implements the `PersistentStorageDelegate` interface with `chip_tool_config.ini` as underlying storage?

Reply:
chip_tool_config.ini is actually an impl that backs PerisstentStorageDelegate.

Reply:
Check the STR that I just posted. Here's the corresponding chip_tool_config.ini
[chip_tool_config.ini.zip](https://github.com/project-chip/connectedhomeip/files/9022825/chip_tool_config.ini.zip)

Reply:
To clarify: the storage state is not self healing

Reply:
We need to regroup on this approach.

This PR attempts to:

1. delete index entries even if data in the other two tables can't be found -> alright, that can make sense; actually, this is a form of partial self-healing.
2. suppress what is believed to be spurious not-found error logging -> I disagree; not-found is a real error and it means the tables are out of sync, or the DeleteAll method is broken; we need actual root cause, or we should preserve the logging.  If we allow (1), logging will only happen once when the index is cleaned.
3. suppress propagating not-found error to callers: again, disagree; this is a real error; the index is supposed to be in sync with the other data in the kvstore; if it's not, the caller should be informed of this.  Again, if we allow (1), this will only happen once when the index is cleaned.

Then, given agreement on 1,2,3, we also need:

* correct code in the PR (it is not quite)
* agreement on other remediations for the underlying design, which is inherently fragile, and from reports, possibly broken

Reply:
Agreed. I would not have used base64 keys though...

Reply:
Fixed the root cause, see https://github.com/project-chip/connectedhomeip/pull/20220. We can close this after that gets merged but we do need *something* because right now CASE resumption does not work after chip-tool restart.